src/elements.jl

```diff
@@ -83,11 +114,17 @@ Element interacting with electrons via a pseudopotential model.
 `key` may be an element symbol (like `:Si`), an atomic number (e.g. `14`)
 or an element name (e.g. `"silicon"`)
 """
-function ElementPsp(key; psp)
-    ElementPsp(periodic_table[key].number, Symbol(periodic_table[key].symbol), psp)
+function ElementPsp(key; psp, use_nlcc=has_density_core(psp))
+    ElementPsp(periodic_table[key].number, Symbol(periodic_table[key].symbol), psp, use_nlcc)
+end
+function ElementPsp(Z::Int, symbol, psp; use_nlcc=has_density_core(psp))
+    ElementPsp(Z, symbol, psp, use_nlcc)
 end
-charge_ionic(el::ElementPsp)   = charge_ionic(el.psp)
+
+charge_ionic(el::ElementPsp) = charge_ionic(el.psp)
 charge_nuclear(el::ElementPsp) = el.Z
+has_density_core(el::ElementPsp) = has_density_core(el.psp)
+use_nlcc(el::ElementPsp) = has_density_core(el.psp) & el.use_nlcc
```

Comment:
I would just return `el.use_nlcc` and rather error on construction if someone uses `use_nlcc=true` for cases where there is no `nlcc`.